src/compiler/transformers/ts.ts

```diff
@@ -909,7 +926,13 @@ namespace ts {
             if (some(staticProperties) || some(pendingExpressions)) {
                 const expressions: Expression[] = [];
                 const isClassWithConstructorReference = resolver.getNodeCheckFlags(node) & NodeCheckFlags.ClassWithConstructorReference;
-                const temp = createTempVariable(hoistVariableDeclaration, !!isClassWithConstructorReference);
+                const temp = createTempVariable(
+                    name => {

```

Comment:
If you make the changes to `visitEachChild` I mentioned above, you shouldn't need this closure as you can pass `addBlockVariableDeclaration` (or whatever we end up calling it) here and set the original node pointer after the fact.